Refactored build script for better error handling

alpaix opened the pull request:
- Replaced throw's with Write-Error's to avoid script termination
- Added -ErrorAction control over build steps execution
- Split build script into granular build steps for better flow
  control
- Split NuGet.Clients build into build and tests msbuild statements

alpaix committed:
Refactored build script for better error handling

+ Replaced throw's with Write-Error's to avoid script termination
+ Added -ErrorAction control over build steps execution
+ Split build script into granular build steps for better flow
control
+ Split NuGet.Clients build into build and tests msbuild statements

alpaix commented:
Default value for -ErrorAction is Continue. To alter the build script behavior add -ErrorAction Stop (or -ea for short).

Valid values:

```
        Continue. Displays the error message and continues executing
        the command. "Continue" is the default value.

        Ignore.  Suppresses the error message and continues
        executing the command. Unlike SilentlyContinue, Ignore
        does not add the error message to the $Error automatic
        variable.

        Inquire. Displays the error message and prompts you for 
        confirmation before continuing execution. This value is rarely
        used.

        SilentlyContinue. Suppresses the error message and continues
        executing the command.

        Stop. Displays the error message and stops executing the
        command.
```

yishaigalatzer commented:
Does it now display the project name that failed? The previous script printed out projectName.

:shipit:

alpaix committed:
Last minute build script fixes

+ Making -Fast parameter control verbosity of nuget.exe and dnu outputs
+ Small fixes

alpaix commented:
It does. Also it repeats all collected errors in build summary.
